packages/browser/src/core/analytics/index.ts

```diff
@@ -524,7 +525,7 @@ export class Analytics
   async ready(
     callback: Function = (res: Promise<unknown>[]): Promise<unknown>[] => res
   ): Promise<unknown> {
-    return Promise.all(
+    return Promise.allSettled(
```

Comment:
Since `allSettled` is a newish browser thing, we might want to try a polyfill
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

```ts
type AllSettledResult<T> = {
  status: 'fulfilled' | 'rejected'
  reason?: Error
  value?: T
}

// promise.allSettled "polyfill"
export const allSettled = <T>(
  promises: Promise<T>[]
): Promise<AllSettledResult<T>[]> => {
  return Promise.all(
    promises.map((promise) => {
      // Per spec, Promise.allSettled([Promise.reject('oops'), 2] ->
      // [{status: "rejected", reason: "oops"}, { "status": "fulfilled", "value": 2 }}
      return Promise.resolve(promise)
        .then(
          (value) =>
            ({
              value,
              status: 'fulfilled',
            } as AllSettledResult<T>)
        )
        .catch(
          (reason) =>
            ({
              reason,
              status: 'rejected',
            } as AllSettledResult<T>)
        )
    })
  )
}



----

import { allSettled } from './all-settled'

describe('all settled', () => {
  it('should return a promise ', async () => {
    const result = allSettled([Promise.resolve('success')])
    expect(result).toBeInstanceOf(Promise)
  })

  it('should handle successes', async () => {
    const result = await allSettled([
      Promise.resolve('success'),
      Promise.resolve('success 2'),
    ])
    const response = [
      { status: 'fulfilled', value: 'success' },
      { status: 'fulfilled', value: 'success 2' },
    ]
    expect(result).toEqual(response)
  })

  it('should handle rejects', async () => {
    const result = await allSettled([
      Promise.reject('oops'),
      Promise.reject('oops 2'),
    ])
    const response = [
      { status: 'rejected', reason: 'oops' },
      { status: 'rejected', reason: 'oops 2' },
    ]
    expect(result).toEqual(response)
  })

  it('should handle mixes of fulfilled/reject', async () => {
    const result = await allSettled([
      Promise.reject('oops'),
      Promise.resolve(2),
    ])
    const response = [
      { status: 'rejected', reason: 'oops' },
      { status: 'fulfilled', value: 2 },
    ]
    expect(result).toEqual(response)
  })

  it('should handle mixes of fulfilled/reject / respect order', async () => {
    const result = await allSettled([
      Promise.resolve(2),
      Promise.reject('oops'),
    ])
    const response = [
      { status: 'fulfilled', value: 2 },
      { status: 'rejected', reason: 'oops' },
    ]
    expect(result).toEqual(response)
  })

  it('should resolve any non-promises passed', async () => {
    const result = await allSettled([2 as any, Promise.reject('oops 2')])
    const response = [
      { status: 'fulfilled', value: 2 },
      { status: 'rejected', reason: 'oops 2' },
    ]
    expect(result).toEqual(response)
  })

  it('should never call catch block', async () => {
    const result = await allSettled([
      Promise.reject('oops'),
      Promise.reject('oops 2'),
    ]).catch(() => {
      throw new Error('fail')
    })
    const response = [
      { status: 'rejected', reason: 'oops' },
      { status: 'rejected', reason: 'oops 2' },
    ]
    expect(result).toEqual(response)
  })
})


```

Reply:
Done

Reply:
@zikaari is this done?

Reply:
I still see that we're not using a polyfill here! I don't think we can get away with using the raw allSettled as it doesn't get added in the normal polyfills.

Reply:
Thanks for pointing out - fixed

Reply:
Can you remind me why this change is needed again? I thought it was originally because we were changing the initialization/ready ordering, but I don't actually see that being the case anymore in the analytics.js loader function.

Our docs currently say this:
> The ready method isn’t invoked if any Destination throws an error (for example, for an expired API key, incorrect settings configuration, or when a Destination is blocked by the browser) during initialization.

So, do we even need this change?

Reply:
Another good catch, we made this change when `ready` could throw - but that was reverted, however this wasn't. I'm gonna revert this back.